### PR TITLE
[MRESOLVER-365] Upgrade dependencies

### PR DESCRIPTION
**Pull request by cstamas**
---

https://issues.apache.org/jira/browse/MRESOLVER-365

### CODE REVIEW

**Comment by cstamas** [comment]
Done, https://github.com/apache/maven-resolver/pull/296

**Review by gnodet — APPROVED** [review]

File: maven-resolver-named-locks-hazelcast/pom.xml [hunk 1]
```diff
@@ -39,21 +39,23 @@
...
       <artifactId>hazelcast</artifactId>
-      <version>5.1.4</version>
+      <version>5.3.0</version>
     </dependency>
     <dependency>
       <groupId>javax.inject</groupId>
       <artifactId>javax.inject</artifactId>
-      <optional>true</optional>
+      <scope>provided</scope>
     </dependency>
```
[the remark this answers is not in the document]
Comment (cstamas) [reply maven-resolver-named-locks-hazelcast/pom.xml:59]: No, just like any maven plugin (w/ these scopes) will not resolve all the needed dependencies outside of Maven. IMHO these artifacts are meant for one thing: to be dropped into `${maven.home}/lib/ext` and nothing more. More cautiously one could say "to be dropped into existing (and working) resolver classpath", but then again scope is right, as "working classpath" of resolver HAS this artifact.

[the remark this answers is not in the document]
Comment (michael-o) [reply maven-resolver-named-locks-hazelcast/pom.xml:59]: But what is the pain then? You create a project with Resolver and Hazelcast named lock, all deps are there, not related to Maven usage. They end up in CP anyway, no?

**Review by michael-o — CHANGES_REQUESTED** [review]

File: maven-resolver-named-locks-hazelcast/pom.xml [hunk 2]
```diff
@@ -39,21 +39,23 @@
   <dependencies>
     <dependency>
       <groupId>org.apache.maven.resolver</groupId>
-      <artifactId>maven-resolver-api</artifactId>
+      <artifactId>maven-resolver-named-locks</artifactId>
+      <scope>provided</scope>
```
[the remark this answers is not in the document]
Comment (cstamas) [reply maven-resolver-named-locks-hazelcast/pom.xml:43]: No it will not, try it out

[the remark this answers is not in the document]
Comment (cstamas) [reply maven-resolver-named-locks-hazelcast/pom.xml:43]: (but i ignore test scope in that output)

**Pull request merged**